packet.go

```diff
@@ -54,31 +54,10 @@ func marshalFileInfo(b []byte, fi os.FileInfo) []byte {
 	// 	   so that number of pairs equals extended_count
 
 	flags, fileStat := fileStatFromInfo(fi)
+	f := newFileAttrFlags(flags)
 
 	b = marshalUint32(b, flags)
```

Comment:
See? Think about what if `flags & sshFileXferAttrExtended == sshFileXferAttrExtended` and `len(fileStat.Extended) == 0`.

We’re going to encode a corrupted packet. 🤦‍♀️

Reply:
Thankfully, that's not actually possible, but why it's not possible is subtle:

```
    // if fi implements FileInfoExtendedData, retrieve extended data from it
    if fiExt, ok := fi.(FileInfoExtendedData); ok {
        fileStat.Extended = fiExt.Extended()
        if len(fileStat.Extended) > 0 {
            flags |= sshFileXferAttrExtended
        }
    }
```

That's from fileStatFromInfo.

So it's not possible to have sshFileXferAttrExtended set while len(fileStat.Extended) == 0.

Reply:
Yes, for `marshalFileInfo` this might be true, but since `FileStat.MarshalTo` would be exported, more people can call it than just this code here, and the `RequestServer` code as well.